recipes/qt/all/conanfile.py

```diff
@@ -0,0 +1,673 @@
+
+        for package in self.deps_cpp_info.deps:
+            args += ["-I " + s for s in self.deps_cpp_info[package].include_paths]
+            args += ["-D " + s for s in self.deps_cpp_info[package].defines]
+            args += ["-L " + s for s in self.deps_cpp_info[package].lib_paths]
+
+        if 'libmysqlclient' in self.deps_cpp_info.deps:
+            args.append("-mysql_config " + os.path.join(self.deps_cpp_info['libmysqlclient'].rootpath, "bin", "mysql_config"))
+        if 'libpq' in self.deps_cpp_info.deps:
+            args.append("-psql_config " + os.path.join(self.deps_cpp_info['libpq'].rootpath, "bin", "pg_config"))
+        if self.settings.os == "Macos":
+            args += ["-no-framework"]
+        elif self.settings.os == "Android":
+            args += ["-android-ndk-platform android-%s" % self.settings.os.api_level]
+            args += ["-android-abis %s" % {"armv7": "armeabi-v7a",
+                                           "armv8": "arm64-v8a",
+                                           "x86": "x86",
+                                           "x86_64": "x86_64"}.get(str(self.settings.arch))]
+
+        if self.settings.get_safe("compiler.libcxx") == "libstdc++":
+            args += ["-D_GLIBCXX_USE_CXX11_ABI=0"]
+        elif self.settings.get_safe("compiler.libcxx") == "libstdc++11":
+            args += ["-D_GLIBCXX_USE_CXX11_ABI=1"]
+
+        if self.options.sysroot:
+            args += ["-sysroot %s" % self.options.sysroot]
+
+        if self.options.device:
+            args += ["-device %s" % self.options.device]
+        else:
+            xplatform_val = self._xplatform()
+            if xplatform_val:
+                if not tools.cross_building(self.settings, skip_x64_x86=True):
+                    args += ["-platform %s" % xplatform_val]
+                else:
+                    args += ["-xplatform %s" % xplatform_val]
+            else:
+                self.output.warn("host not supported: %s %s %s %s" %
+                                 (self.settings.os, self.settings.compiler,
+                                  self.settings.compiler.version, self.settings.arch))
+        if self.options.cross_compile:
+            args += ["-device-option CROSS_COMPILE=%s" % self.options.cross_compile]
+
+        def _getenvpath(var):
+            val = os.getenv(var)
+            if val and tools.os_info.is_windows:
+                val = val.replace("\\", "/")
+                os.environ[var] = val
+            return val
+
+        value = _getenvpath('CC')
+        if value:
+            args += ['QMAKE_CC="' + value + '"',
+                     'QMAKE_LINK_C="' + value + '"',
+                     'QMAKE_LINK_C_SHLIB="' + value + '"']
+
+        value = _getenvpath('CXX')
+        if value:
+            args += ['QMAKE_CXX="' + value + '"',
+                     'QMAKE_LINK="' + value + '"',
+                     'QMAKE_LINK_SHLIB="' + value + '"']
```

Comment:
This doesn't work for msvc-like compilers, e.g. `CC=cl` or `CC=clang-cl` or `CC=icl`. It is quite uncommon to invoke the linker through these compilers, so I doubt that qt works with anything other than a path to the linker.

Given that the environment variables `CC` and `CXX` don't actually correspond to the linker, I think the best solution here is to decouple the two and just set `QMAKE_CC` and `QMAKE_CXX`. You could add something like checking for the `LD` environment variable to set the linker if that functionality is still desired.